src/core/public/chrome/nav_group/nav_group_service.ts

```diff
@@ -159,15 +197,81 @@ export class ChromeNavGroupService {
   }
   async start({
     navLinks,
+    application,
   }: {
     navLinks: ChromeNavLinks;
+    application: InternalApplicationStart;
   }): Promise<ChromeNavGroupServiceStartContract> {
     this.navLinks$ = navLinks.getNavLinks$();
+
+    const currentNavGroupId = sessionStorage.getItem(CURRENT_NAV_GROUP_ID);
+    this.currentNavGroup$ = new BehaviorSubject<ChromeNavGroup | undefined>(
+      currentNavGroupId ? this.navGroupsMap$.getValue()[currentNavGroupId] : undefined
+    );
+
+    const setCurrentNavGroup = (navGroupId: string | undefined) => {
+      const navGroup = navGroupId ? this.navGroupsMap$.getValue()[navGroupId] : undefined;
+      if (navGroup && navGroup.status !== NavGroupStatus.Hidden) {
+        this.currentNavGroup$.next(navGroup);
+        sessionStorage.setItem(CURRENT_NAV_GROUP_ID, navGroup.id);
+      } else {
+        this.currentNavGroup$.next(undefined);
+        sessionStorage.removeItem(CURRENT_NAV_GROUP_ID);
+      }
+    };
+
+    // erase current nav group when switch to home page
+    application.currentAppId$.subscribe((appId) => {
+      if (appId === 'home') {
+        setCurrentNavGroup(undefined);
+      }
+    });
+
     return {
       getNavGroupsMap$: () => this.getSortedNavGroupsMap$(),
       getNavGroupEnabled: () => this.navGroupEnabled,
+
+      getCurrentNavGroup$: () => this.currentNavGroup$,
+      setCurrentNavGroup,
+
+      prependCurrentNavGroupToBreadcrumbs: (breadcrumbs: ChromeBreadcrumb[]) => {
```

Comment:
Shall we add this kind of UI logic into header_breadcrumbs.tsx and make nav_group_service a pure data service?